Sources/Testing/ABI/EntryPoints/EntryPoint.swift

```diff
@@ -590,32 +590,40 @@ public func configurationForEntryPoint(from args: __CommandLineArguments_v0) thr
 /// specified ABI version.
 ///
 /// - Parameters:
-///   - version: The ABI version to use.
+///   - version: The numeric value of the ABI version to use.
 ///   - encodeAsJSONLines: Whether or not to ensure JSON passed to
 ///     `eventHandler` is encoded as JSON Lines (i.e. that it does not contain
 ///     extra newlines.)
-///   - eventHandler: The event handler to forward encoded events to. The
+///   - targetEventHandler: The event handler to forward encoded events to. The
 ///     encoding of events depends on `version`.
 ///
 /// - Returns: An event handler.
 ///
 /// - Throws: If `version` is not a supported ABI version.
 func eventHandlerForStreamingEvents(
-  version: Int?,
+  version versionNumber: Int?,
   encodeAsJSONLines: Bool,
-  forwardingTo eventHandler: @escaping @Sendable (UnsafeRawBufferPointer) -> Void
+  forwardingTo targetEventHandler: @escaping @Sendable (UnsafeRawBufferPointer) -> Void
 ) throws -> Event.Handler {
-  switch version {
+  func eventHandler(for version: (some ABI.Version).Type) -> Event.Handler {
+    return version.eventHandler(encodeAsJSONLines: encodeAsJSONLines, forwardingTo: targetEventHandler)
+  }
+
+  return switch versionNumber {
```

Comment:
We could refactor this `switch` to produce an `any ABI.Version.Type`, or even move it into `ABI` as `ABI.version(withVersionNumber:)`, but that would require returning an existential `any` which would preclude its eventual use in Embedded Swift.